lib/mllp/receiver.ex

```diff
@@ -186,27 +173,41 @@ defmodule MLLP.Receiver do
       raise "The packet framer module #{packet_framer_mod} does not implement the MLLP.Dispatcher behaviour"
 
     receiver_id = Keyword.get(opts, :ref, make_ref())
-    transport_mod = :ranch_tcp
 
-    transport_opts = Keyword.get(opts, :transport_opts, default_transport_opts())
+    transport_opts =
+      default_transport_opts()
+      |> Map.merge(Keyword.get(opts, :transport_opts, %{}))
+
+    tls = Map.has_key?(transport_opts, :tls)
+    transport_mod = if tls, do: :ranch_ssl, else: :ranch_tcp
+
+    {tls_options, transport_opts1} = Map.pop(transport_opts, :tls, [])
```

Comment:
There's a lot to unpack as far as tls options go. I think we should provide sensible and secure defaults, `verify_peer` should be one as an example. There's a lot to review in this area, such that I wouldn't ask to do it in this PR. I think we should tighten up in another PR, but wanted to simply call it out.

Reply:
I can add `verify_peer` as default option, when `tls` options are provided. I did look up at other options (https://www.erlang.org/doc/man/ssl.html#type-server_option), but didn't see a need to set default for those other options because `:ssl` module set them default anyway.

Reply:
Yeah, let's dig in outside of this PR IMO. Specifically around https://erlef.github.io/security-wg/secure_coding_and_deployment_hardening/ssl

Reply:
I think some of these options are more applicable to ssl client side when connecting to a sender. In our example, it's our sender module. For listener, our options should be good unless we are also adding client certificate validation which we will do as a separate PR. One interesting option is depth, which I looked up here https://www.erlang.org/doc/man/ssl.html#type-common_option and it looks like default value is 10 for this, which seems odd.